api/v1alpha1/argocd_types.go

```diff
@@ -960,10 +960,12 @@ func (argocd *ArgoCD) IsDeletionFinalizerPresent() bool {
 	return false
 }
 
-// WantsAutoTLS returns true if user configured a route with reencryption
-// termination policy.
+// WantsAutoTLS returns true if:
+// 1. user has configured a route with reencrypt.
+// 2. user has not configured TLS and we default to reencrypt.
 func (s *ArgoCDServerSpec) WantsAutoTLS() bool {
-	return s.Route.TLS != nil && s.Route.TLS.Termination == routev1.TLSTerminationReencrypt
+	return s.Route.TLS == nil ||
+		(s.Route.TLS != nil && s.Route.TLS.Termination == routev1.TLSTerminationReencrypt)
```

Comment:
Do we need the extra not nil check `s.Route.TLS != nil` ? Since `nil` check is already done, it is implied that the other condition would be evaluated only if its not nil. Is my understanding correct or Did I misunderstood the logic?

Reply:
You are right, Anand. Updated the logic.